src/api/pubsub.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const promisify = require('promisify-es6')
+const PubsubMessageStream = require('../pubsub-message-stream')
+const stringlistToArray = require('../stringlist-to-array')
+
+/* Internal subscriptions state and functions */
+let subscriptions = {}
+
+const addSubscription = (topic, request) => {
+  subscriptions[topic] = { request: request }
+}
+
+const removeSubscription = promisify((topic, callback) => {
+  if (!subscriptions[topic]) {
+    return callback(new Error(`Not subscribed to ${topic}`))
+  }
+
+  subscriptions[topic].request.abort()
+  delete subscriptions[topic]
+
+  if (callback) {
+    callback(null)
+  }
+})
+
+/* Public API */
+module.exports = (send) => {
+  return {
+    subscribe: promisify((topic, options, callback) => {
+      const defaultOptions = {
+        discover: false
+      }
+
+      if (typeof options === 'function') {
+        callback = options
+        options = defaultOptions
+      }
+
+      if (!options) {
+        options = defaultOptions
+      }
+
+      // If we're already subscribed, return an error
+      if (subscriptions[topic]) {
+        return callback(new Error(`Already subscribed to '${topic}'`))
+      }
```

Comment:
If we have several subscriptions, we just need to feed the messages to all of them.